### PR TITLE
Climber (v1)

### PR DESCRIPTION
**Pull request by NWSchallock**
PTAL

### CODE REVIEW

File: src/OI.cpp [hunk 1]
```diff
@@ -61,6 +61,14 @@ OI::OI()
   dLB.WhenPressed(new SetIntake(1.0));
   dRB.WhenPressed(new SetIntake(0.0));
 
+  /*
+   *dX.WhenPressed(new SetClimber(1.0));
+   *dY.WhenPressed(new SetClimber(-1.0));
+   *dB.WhenPressed(new SetClimber(0.0));
+   */
```
Comment (duane701) [inline src/OI.cpp]: Do or do not. Either put these in or delete them.

File: src/Subsystems/Climber.cpp [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+std::shared_ptr<Climber> Climber::self;
+std::shared_ptr<Climber> Climber::getInstance() {
+  if (! self) {
+    self = std::shared_ptr<Climber>(new Climber);
+  }
+  return self;
+}
+
+Climber::Climber() : Subsystem("Climber"),
```
Comment (duane701) [inline src/Subsystems/Climber.cpp]: Replace `"Climber"` with `kSubsystemName`

File: src/Subsystems/Climber.cpp [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+	// SetDefaultCommand(new MySpecialCommand());
+  SetDefaultCommand(new SetClimber(0.0));
+}
+
+// Put methods for controlling this subsystem
+// here. Call these from Commands.
+
+void Climber::SetClimberMove(double speed){
+  leftClimber.Set(speed);
+  rightClimber.Follow(leftClimber);
```
Comment (duane701) [inline src/Subsystems/Climber.cpp]: `Follow` does not go here! `Follow` goes in the constructor (`Climber::Climber`)

File: src/Subsystems/Climber.cpp [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+void Climber::InitDefaultCommand() {
+	// Set the default command for a subsystem here.
+	// SetDefaultCommand(new MySpecialCommand());
+  SetDefaultCommand(new SetClimber(0.0));
+}
+
+// Put methods for controlling this subsystem
+// here. Call these from Commands.
+
+void Climber::SetClimberMove(double speed){
```
Comment (duane701) [inline src/Subsystems/Climber.cpp]: Using `Climber` in a `Climber` interface is redundant. e.g. `climber.SetClimberMove()`. How about changing `SetClimberMove` to `Climb`.

File: src/Subsystems/Climber.h [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+  WPI_TalonSRX leftClimber;
+  WPI_TalonSRX rightClimber;
+
+public:
+  static std::shared_ptr<Climber> getInstance();
+
+
+	Climber();
+	void InitDefaultCommand() override;
+	void SetClimberMove(double speed);
```
Comment (duane701) [inline src/Subsystems/Climber.h]: See comment. Change `SetClimberMove` to `Climb`.

**Review by duane701 — CHANGES_REQUESTED** [review]
FTP

**Comment by NWSchallock** [comment]
PTAL

File: src/Subsystems/Climber.cpp [hunk 6]
```diff
@@ -18,10 +18,11 @@ std::shared_ptr<Climber> Climber::getInstance() {
   return self;
 }
 
-Climber::Climber() : Subsystem("Climber"),
+Climber::Climber() : Subsystem("kSusystemName"),
```
Comment (duane701) [inline src/Subsystems/Climber.cpp]: Umm, let's try this again. Should be

    Climber::Climber() : Subsystem(kSubsystemName),

File: src/Commands/SetClimber.cpp [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+}
+
+// Called just before this Command runs the first time
+void SetClimber::Initialize() {
+
+}
+
+// Called repeatedly when this Command is scheduled to run
+void SetClimber::Execute() {
+  Climber::getInstance()->SetClimberMove(mSpeed);
```
Comment (duane701) [inline src/Commands/SetClimber.cpp]: `SetClimberMove()` no longer exists. Did you build this code?

**Review by duane701 — CHANGES_REQUESTED** [review]
FTP

**Comment by NWSchallock** [comment]
PTAL

File: src/OI.h [hunk 8]
```diff
@@ -40,7 +40,7 @@ class OI {
   double getCoDriverRightXAxis() const;
   double getCoDriverRightYAxis() const;
 
-  bool isDriverBPressed() const;
+  bool isDriverBPressed();
```
Comment (duane701) [inline src/OI.h]: Bummer

**Review by duane701 — APPROVED** [review]
LGTM

**Pull request merged**